h2o-py/h2o/explanation/_explain.py

```diff
@@ -1305,35 +1308,59 @@ def ice_plot(
             factor_map = _factor_mapper(NumpyFrame(frame[column]).from_factor_to_num(column))
 
         deciles = [int(round(frame.nrow * dec / 10)) for dec in range(11)]
+        deciles[10] = frame.nrow - 1
         colors = plt.get_cmap(colormap, 11)(list(range(11)))
         plt.figure(figsize=figsize)
         for i, index in enumerate(deciles):
+            percentile_string = "{}th Percentile".format(i * 10)
             tmp = NumpyFrame(
                 model.partial_plot(
                     frame,
                     cols=[column],
                     plot=False,
                     row_index=index,
                     targets=target,
-                    nbins=20 if not is_factor else 1 + frame[column].nlevels()[0]
+                    nbins=100 if not is_factor else 1 + frame[column].nlevels()[0],
+                    include_na=True
                 )[0]
             )
+            user_splits = dict()
+            orig_value = frame.as_data_frame()[column][index]
             encoded_col = tmp.columns[0]
+            if _isnan(orig_value):
+                id = np.where(tmp[encoded_col] == tmp.from_factor_to_num(encoded_col)['.missing(NA)'])[0][0] if is_factor else np.where(np.isnan(tmp[encoded_col]))[0][0]
+                orig_null_value = tmp.from_num_to_factor(encoded_col)[tmp[encoded_col][id]] if is_factor else tmp[encoded_col][id]
+                interval = "[{}, {}]".format(orig_null_value,  tmp["mean_response"][id])
+                warnings.warn("Original observation of \"" + encoded_col + "\" for " + percentile_string + " is " + interval + ". Ploting of NAs is not yet supported." )
+            else:
+                user_splits[column] = [orig_value]
+                orig_tmp = NumpyFrame(
+                    model.partial_plot(
+                        frame,
+                        cols=[column],
+                        plot=False,
+                        row_index=index,
+                        targets=target,
+                        user_splits=user_splits,
+                    )[0]
+                )
+                plt.scatter(orig_tmp[encoded_col], orig_tmp["mean_response"],
+                            color=[colors[i]], marker='o', s=150, alpha=0.5)
```

Comment:
It would be good to have the "original observation" entry in the legend as is done in R.

Reply:
[93d3066](https://github.com/h2oai/h2o-3/pull/6020/commits/93d3066903cc11f3d2
![cncnfare](https://user-images.githubusercontent.com/18303397/152797385-eb0e3646-a039-45f7-b18c-9fe7e507a735.png)
d52cb61d55cd07fb810d12)
![dsdscabin_type](https://user-images.githubusercontent.com/18303397/152797407-0bb18624-3386-4a90-ba94-1260bf3cecf0.png)